compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -80,6 +94,35 @@
                                     PlatformLibraryScope.DEFAULT,
                                     getPackageName(rootPackageContext)));
 
+        // TODO: Move to a compiler extension once Compiler revamp is complete
+        // 4) Add the Observability Symbols Jar
+        if (rootPackageContext.compilationOptions().observabilityIncluded()) {
+            Manifest manifest = new Manifest();
+            manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
+
+            try {
+                // Generating an empty Jar which can be used by the Observability Symbol Collector
+                String packageName = getPackageName(rootPackageContext);
+                Path jarPath = Path.of(System.getProperty("java.io.tmpdir"),
```

Comment:
You can use Files.createTempFile here. Also can we create this in target so we can use it for caching ?

Reply:
Shall we move the jar creation to [ProjectUtils](https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java) class?

Reply:
@hevayo ATM persisting the thin jars is handled by the `CompilationCache` and there is no way to check if observability is enabled in the CompilationCache class. 
https://github.com/ballerina-platform/ballerina-lang/blob/0ed6999d21d053407ab3759b8e8ddf3cc9431364/compiler/ballerina-lang/src/main/java/io/ballerina/projects/repos/FileSystemCache.java#L100-L102

Another way is to use the `Target` model but that would need a refactor to include the package descriptor info in our `Target` model since the jar caches path needs the `org`, `package name`, `version`, and `target platform` to create the directory structure. So I think having it in a temp location would suffice for now.

Reply:
Moved to ProjectUtils and changed the temp file creation to use `Files.createTempFile()` in b49b3969bdad31895d83daa58ac52f03e03524df